instance/tests/models/factories/server.py

```diff
@@ -0,0 +1,131 @@
+# -*- coding: utf-8 -*-
+#
+# OpenCraft -- tools to aid developing and hosting free software projects
+# Copyright (C) 2015 OpenCraft <xavier@opencraft.com>
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+"""
+OpenStackServer model - Factories
+"""
+
+# Imports #####################################################################
+
+import factory
+from factory.django import DjangoModelFactory
+from functools import wraps
+from mock import MagicMock, Mock, patch
+
+from instance.models.server import OpenStackServer
+from instance.tests.base import add_fixture_to_object
+from instance.tests.models.factories.instance import OpenEdXInstanceFactory
+
+
+# Functions ###################################################################
```

Comment:
@dachary I added an integration test, testing a full provisioning of an edX instance, including VM creation & ansible run. Cf https://github.com/open-craft/opencraft/commit/e095d89f657830a98120b501c2c1e5c3e70e16ef

Since running them require access to a working OpenStack cloud, they are run optionally by `make test`, when the config variables are available. I have setup the CircleCI build to automatically run them on PR builds - it's really neat actually, it does a lot of testing automatically now, with only a few additional lines of code : ) It will likely help preventing the type of regressions I introduced this week, where the test suite was running fine, but the edX instances builds were failing.

Sample build with integration tests: https://circleci.com/gh/open-craft/opencraft/27

TODO: Get the build to fail when the ansible run fails - currently the instance is considered provisioned, regardless of the return value from ansible. This will likely require to also add an error state for VMs, and maybe retries if random ansible failures prove to be an issue.